### PR TITLE
oe fuzzer target programs and onefuzz workflow

### PR DESCRIPTION
**Pull request by mrragava**
Open Encalve fuzzing and onefuzz workflow.
- openenclave fuzzer support library
- fuzzer target programs for host and enclave apis
- onefuzz workflow and scripts

Signed-off-by: Ragavan Dasarathan <mrragava@gmail.com>

### CODE REVIEW

**Comment by ghost** [comment]
@mrragava: GitHub didn't allow me to assign the following users: swamymsft.

Note that only [openenclave members](https://github.com/orgs/openenclave/people), repo collaborators and people who have commented on this issue/PR can be assigned. Additionally, issues/PRs can only have 10 assignees at the same time.
For more information please see [the contributor guide](https://git.k8s.io/community/contributors/guide/first-contribution.md#issue-assignment-in-github)

<details>

In response to [this](https://github.com/openenclave/openenclave-security/pull/8#issuecomment-883001349):

>/assign @swamymsft


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: build.sh
Comment (douglasmaciver) [inline build.sh:146]: Shouldn't the fuzzer targets be built with the expected optimization level?

**Review by swamymsft — APPROVED** [review]

**Comment by ghost** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/openenclave/openenclave-security/pull/8#" title="Author self-approved">mrragava</a>*, *<a href="https://github.com/openenclave/openenclave-security/pull/8#pullrequestreview-711130069" title="Approved">swamymsft</a>*

The full list of commands accepted by this bot can be found [here](?repo=openenclave%2Fopenenclave-security).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/openenclave/openenclave-security/blob/main/OWNERS)~~ [mrragava]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: build.sh
Comment (mrragava) [reply build.sh:146]: Optimization levels has to be disabled for LLVM sanitizers to instrument the edges and to insert callbacks in IR pass.

Comment (douglasmaciver) [reply build.sh:146]: This is unfortunate given that optimizations tend to remove some security protections.

**Pull request merged**
